helix-common/src/main/java/org/apache/helix/constants/InstanceConstants.java

```diff
@@ -8,4 +8,12 @@ public enum InstanceDisabledType {
     USER_OPERATION,
     DEFAULT_INSTANCE_DISABLE_TYPE
   }
+
+  public enum InstanceOperation {
+    EVACUATE,
+    SWAP_IN,
+    SWAP_OUT,
+    ENABLE,
```

Comment:
Just curious, do you think this might create confusion with Instance config HELIX_ENABLED. 
I remember you mentioned SWAPED_IN, this transitioning to `instanceConfig.setInstanceEnabled` also looks good.   

`Enable` is quite  common term and maybe this way, from the name itself its clear that which state lifecycle we are talking about.

Reply:
This field is an Union of all operation. Including enable and disable. We will also update  `HELIX_ENABLED ` when operation is set to enable/disable.
We decide not to introduce life cycle as it adds more complexity.

Reply:
Sounds good, thanks @xyuanlu!